src/main/java/net/minecraftforge/event/entity/player/PlayerInteractEvent.java

```diff
@@ -322,6 +322,28 @@ public LeftClickEmpty(Player player)
         }
     }
 
+    /**
+     * This event is fired when an item would be used via {@link Item#useOn}<br>
+     * It allows for the usage to be changed or cancelled.<br>
+     * This event allows finer-tuned control over the actual item usage that cannot be achieved by using {@link RightClickBlock}
+     */
+    @Cancelable
+    public static class ItemUseEvent extends PlayerInteractEvent
+    {
+        protected final UseOnContext ctx;
```

Comment:
Why is this field `protected` rather than `private`?

Reply:
I just tend to use protected over private unless it explicitly needs to be private.  I can swap it, since forge uses private over protected by default